tests/Circle3d.elm

```diff
@@ -22,6 +25,49 @@ jsonRoundTrips =
         Decode.circle3d
 
 
+throughPoints : Test
+throughPoints =
+    Test.describe "throughPoints"
+        [ Test.fuzz3 Fuzz.point3d
+            Fuzz.point3d
+            Fuzz.point3d
+            "All given points lie on the circle constructed using `throughPoints`"
+            (\p1 p2 p3 ->
+                let
+                    -- if the first three points are collinear it is not possible to construct a circle
+                    -- that passes through them
+                    isValidInput =
```

Comment:
I think this could be done more succinctly and robustly by a triangle area check - if the area of the triangle formed by the three points is zero, then the points are collinear. Probably also check for area being below a tolerance (say 1e-6) instead of exactly zero, to filter out really degenerate cases where there will be bad roundoff.

Reply:
Done. 

Out of interest I set `isValidInput = True` and it seems the fuzzer pretty much never creates three collinear  points. Maybe we should investigate the fuzzer and how to get better "coverage" a little bit in the future. An easy solution would be to just always do both: fuzz based testing and testing using hard coded test cases.